Feature/#3

58peiShan opened the pull request:
1.會員功能，登入註冊修改  
**資料庫database表格欄位有修改，已上傳到feature#3分支**

Commit:
#3
1.會員登入註冊，修改資料

58peiShan committed:
Add files via upload

58peiShan committed:
Update and rename artwork (3).sql to artwork.sql

Commit:
#3
改一下註冊流程

Commit:
Merge branch 'feature/#3' of https://github.com/58peiShan/FEproject into feature/#3

Commit:
#3
修正UC瀏覽器不支援的語法(window.scrollTo)

panchinpeng commented on frontend/src/pages/user/dashboard.jsx:
```diff
@@ -0,0 +1,133 @@
+            <input
+              className="nameEdit"
+              ref={nameRef}
+              placeholder={userdata.userName ? userdata.userName : "none"}
```
![image](https://user-images.githubusercontent.com/16019808/192200572-0439e859-437e-4890-a218-29d030206c2a.png)
已登入，卻顯示none

panchinpeng commented on frontend/src/pages/user/dashboard.jsx:
```diff
@@ -0,0 +1,133 @@
+    }, [token]);
+
+    const editName = (e) => {
+      if (e.key === "Enter") {
```
修改username後，按下enter後，資料無法正常設定

https://user-images.githubusercontent.com/16019808/192201538-2b7f4cc8-2e73-4600-b5bd-1fba9003ea91.mp4

panchinpeng commented on frontend/src/pages/home/home.jsx:
```diff
@@ -2,23 +2,41 @@ import { Routes, Route, Link } from "react-router-dom";
+              justifyContent: "space-between",
+            }}
+          >
             <div className="imgContainer">
```
UC 首頁跑版拉
![image](https://user-images.githubusercontent.com/16019808/192202173-d41a32fc-bdf2-479a-b1ea-f2c70b9c1d42.png)
![image](https://user-images.githubusercontent.com/16019808/192202263-778de4ff-1079-420a-a2e2-b0b253b0ebe3.png)

panchinpeng commented on frontend/src/component/Header.jsx:
```diff
@@ -1,38 +1,80 @@
+    }
+  };
+  const logoutHandler = () => {
+    const logout = confirm("確定要登出嗎？");
```
登出後購物車的數量不會清空

https://user-images.githubusercontent.com/16019808/192203484-40b1ee1a-5484-4e38-a0df-d67a7fc6eb36.mp4

panchinpeng reviewed (commented):
#3 CR問題

Commit:
#3
1.會員登入後修改功能修正（請先清除localStorage）
2.會員註冊密碼雜湊後再儲存（請匯入我更新於#3的DB）
3.商品個別詳細資訊
4.paypal結帳
5.結帳後商品展示在個人頁面(購買清單)，並在商品頁面顯示SOLD OUT
6.首頁串接TokenViewAPI顯示最新10筆NFT交易，collection（該作品系列）地址連外部連結看詳細資訊

58peiShan committed:
0929, update

0929, 更改原先儲存密碼

58peiShan committed:
Update README.md

panchinpeng commented:
![image](https://user-images.githubusercontent.com/16019808/193210800-8b155d8d-59e0-4ef7-b275-8103018e793d.png)

登入時，server出現錯誤如圖

panchinpeng commented:
https://user-images.githubusercontent.com/16019808/193210941-939aa524-da83-4461-acd9-7c4c1d29fdb4.mp4

帳號超過6碼，瘋狂打Api

58peiShan committed:
Delete artwork.sql

Commit:
#3
1.修正結帳金額為乙太幣對美元後價格

panchinpeng commented on frontend/src/pages/products/productItem.jsx:
```diff
@@ -0,0 +1,72 @@
+  const handleBuy = (id, amount) => {
+    if (!localStorage.getItem("auth")) {
+      alert("請先登入！");
+      return;
```
return 可省略

panchinpeng commented on frontend/src/pages/products/productItem.jsx:
```diff
@@ -0,0 +1,72 @@
+      alert("請先登入！");
+      return;
+    } else if (purchaseList.includes(id) || amount === 0) {
+      return;
```
return 可省略

panchinpeng commented on frontend/src/pages/products/productItem.jsx:
```diff
@@ -0,0 +1,72 @@
+      .then((data) => setItem(...data));
+  }, []);
+  return (
+    <>
```
<>可省略

panchinpeng commented on frontend/src/actions/cartListAction.js:
```diff
@@ -0,0 +1,39 @@
+          //  a.push(...data);
+          dispatch(getPurchaseSuccess(data));
+        })
+        .then()
```
可移除

panchinpeng commented on frontend/src/pages/user/SignIn.jsx:
```diff
@@ -125,18 +142,26 @@ class SignIn extends Component {
-            <Link to="/user/login">已有帳號，登入</Link>
+
+          <div className="">
+            <button className="btnSec">
```
點擊按鈕附近一直重整，沒辦法跳頁

https://user-images.githubusercontent.com/16019808/193519841-12447d8c-2bcd-4cc3-bc08-bc79a10aa4a7.mp4

58peiShan replied:
已修正，因未取消預設行為

panchinpeng reviewed (commented):
10/03

Commit:
#3
修正登入/註冊切換時的重整行為

Commit:
#3
1.RWD切版
2.兌美元金額整理至reduxStore

58peiShan committed:
Merge branch 'develop' into feature/#3
